### PR TITLE
Add Conditional Content UI 

### PR DESCRIPTION
**Pull request by chrispymm**
This PR adds in the Controllers and UI for conditional content areas.

However it does not yet use a version of the metadata presenter that supports conditional components.  

All of the UI and behaviours are behind a flag so nothing is exposed to the user, and is safe to ship.

### CODE REVIEW

File: acceptance/support/capybara.rb
Comment (chrispymm) [inline acceptance/support/capybara.rb:1]: This fixes the Deprecation warning when running acceptance tests.

Means that we can bump rspec/selenium - as the latest version enforces the deprecation, so failed pipelines and we had to revert.

File: app/javascript/src/index.js [hunk 1]
```diff
@@ -89,8 +98,8 @@ switch(controllerAndAction()) {
     break;
 
   default:
-       console.log(controllerAndAction());
-       Controller = DefaultController;
+    //console.log(controllerAndAction());
```
Comment (StevenLeighton21) [inline app/javascript/src/index.js:101]: cheeky comment

File: app/services/content_visibility_creation.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  attr_accessor :conditional_content, :latest_metadata
+
+  def save
+    return false if conditional_content.invalid? || conditional_content.any_errors?
+
+    create_version
+  end
+
+  def metadata
+    # conditional_content.conditionals.map{|c| c.to_metadata }
```
Comment (StevenLeighton21) [inline app/services/content_visibility_creation.rb:16]: cheeky comment :)

File: spec/models/conditional_content_spec.rb [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+  let(:previous_page) do
+    service.find_page_by_url('pets')
+  end
+  let(:previous_flow_uuid) do
+    previous_page.uuid
+  end
+  let(:current_page) do
+    service.find_page_by_url('furry')
+  end
+  # let(:conditional_content_uuid) { 'ac914bb1-c4b8-44a0-bacd-2d23a6597f1a' }
```
Comment (StevenLeighton21) [inline spec/models/conditional_content_spec.rb:22]: 👀

File: app/javascript/src/index.js (hunk 1 above)
Comment (chrispymm) [reply app/javascript/src/index.js:101]: yeah, I lef tthis deliberately as its helpful to just uncomment when you need debugging

**Review by StevenLeighton21 — APPROVED** [review]

**Review by H3ll3m4 — APPROVED** [review]
🚀

**Pull request merged**